src/clr-core/button/base-button.element.scss

```diff
@@ -0,0 +1,224 @@
+// The full license information can be found in LICENSE in the root directory of this project.
+
+@import './../styles/tokens/generated/index';
+@import './../styles/mixins/utils';
+@import './../styles/mixins/mixins';
+
+// TODO: progress spinner CSS deprecated, will be replaced by cds-circular-progress
+@import './../../clr-angular/progress/spinner/variables.spinner';
+@import './../../clr-angular/image/icons.clarity';
+@import './../../clr-angular/progress/spinner/spinner.clarity';
+
+:host {
+  --box-shadow-color: #{$cds-token-color-action-900};
+
+  --border-radius: #{$cds-token-global-border-radius};
+  --border-width: #{$cds-token-global-border-width};
+  --border-color: #{$cds-token-color-action-600};
+  --background: #{$cds-token-color-action-600};
+
+  --color: #{$cds-token-color-neutral-0};
+  --font-size: #{$cds-token-space-size-6};
+
+  --padding: #{$cds-token-layout-space-sm};
+  --height: #{$cds-token-space-size-11};
+
+  display: inline-block;
+
+  // following is a fix for Firefox cutting off right edge of buttons. remove when no longer needed.
+  // it is okay to force this to 1px because we never want it to grow or be customized
+  padding-right: 1px;
+}
+
+.private-host {
+  -webkit-appearance: none !important;
+  align-items: center;
+  background: var(--background);
+  border-color: var(--border-color);
+  border-radius: var(--border-radius);
+  border-style: solid;
+  border-width: var(--border-width);
+  color: var(--color);
+  cursor: pointer;
+  display: inline-flex;
+  font-size: var(--font-size);
+  height: var(--height);
+  justify-content: center;
+  line-height: 1em;
+  min-width: #{$cds-token-space-size-13};
+  overflow: visible;
+  padding: var(--padding);
+  position: relative;
+  text-align: center;
+  text-decoration: none;
+  text-overflow: ellipsis;
+  transform: translateZ(0px); // for chrome rendering bug
+  user-select: none;
+  vertical-align: middle;
+  white-space: nowrap;
+
+  &:active {
+    box-shadow: 0 #{$cds-token-space-size-1} 0 0 var(--box-shadow-color) inset;
+  }
+
+  & > span {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: var(--font-size);
+  }
+
+  .button-status-icon {
+    transform: translateY(calc(-1 * #{$cds-token-space-size-1}));
+
+    cds-icon {
+      @include min-equilateral(#{$cds-token-space-size-8});
+      fill: var(--color);
+    }
+
+    cds-icon[shape='error-standard'] {
+      --color: #{$cds-token-color-danger-700};
+    }
+  }
+}
+
+:host(:active) .private-host {
+  box-shadow: 0 #{$cds-token-space-size-1} 0 0 var(--box-shadow-color) inset;
+}
+
+:host(:hover) {
+  --background: #{$cds-token-color-action-800};
+}
+
+::slotted {
+  line-height: 1em;
+  color: inherit;
+}
+
+::slotted(a) {
+  text-decoration: none !important;
+  display: block;
+  height: 100%;
+  color: inherit;
+}
+
+::slotted(cds-icon) {
+  --color: inherit;
+
+  @include equilateral(calc(var(--font-size) + (2 * #{$cds-token-space-size-2})));
```

Comment:
Icons in buttons are sized up a smidge and then readjusted with a negative top margin. This allows the actual glyph to be "16px".